### PR TITLE
Lag in frames option

### PR DESCRIPTION
**Pull request by Noarkhh**

### CODE REVIEW

**Comment by Noarkhh** [comment]
I believe a better way to handle this keyframes test would be to use manual demands to ensure that a new frame is demanded only once the previous is processed, so that we could ensure that the events actually arrive in between the buffers

File: lib/membrane_vpx/encoder/vp8_encoder.ex [hunk 1]
```diff
@@ -36,6 +36,20 @@ defmodule Membrane.VP8.Encoder do
                 of the incoming stream. Some reference recommended bitrates can be also found
                 [here](https://support.google.com/youtube/answer/1722171#zippy=%2Cbitrate)
                 """
+              ],
+              g_lag_in_frames: [
+                spec: non_neg_integer(),
+                default: 5,
```
Comment (varsill) [inline lib/membrane_vpx/encoder/vp8_encoder.ex:42]: is this one the enoder's default or was it just your choice?

File: lib/membrane_vpx/encoder/vpx_encoder.ex [hunk 2]
```diff
@@ -128,17 +132,18 @@ defmodule Membrane.VPx.Encoder do
         {fixed_deadline, _framerate} -> fixed_deadline |> Membrane.Time.as_microseconds(:round)
       end
 
-    rc_target_bitrate = get_target_bitrate(state.rc_target_bitrate, width, height, framerate)
+    user_encoder_config =
+      process_user_encoder_config(state.user_encoder_config, width, height, framerate)
```
Comment (varsill) [inline lib/membrane_vpx/encoder/vpx_encoder.ex:136]: Couldn't we store the already processed `user_encoder_config` in the element's state?

**Review by varsill — COMMENTED** [review]
How about adding a single option in the element, common for all the VPx encoder's settings mapping?

File: lib/membrane_vpx/encoder/vpx_encoder.ex (hunk 2 above)
Comment (Noarkhh) [reply lib/membrane_vpx/encoder/vpx_encoder.ex:136]: we could, but the target bitrate needs to be recalculated on new stream format either way, so i don't think there's much difference

File: lib/membrane_vpx/encoder/vp8_encoder.ex (hunk 1 above)
Comment (Noarkhh) [reply lib/membrane_vpx/encoder/vp8_encoder.ex:42]: my choice, the default was 25

**Review by varsill — COMMENTED** [review]
How about adding a test to check if `g_lag_in_frames` works? I believe it's quite a crucial feature. All I would like to check would be to use `g_lag_in_frames: 0` and make sure that no frames are buffered

**Review by varsill — APPROVED** [review]
🥇

**Pull request merged**